cmake/CPM.cmake

```diff
@@ -0,0 +1,19 @@
+set(CPM_DOWNLOAD_VERSION 0.27.3)
+
+if(CPM_SOURCE_CACHE)
+  set(CPM_DOWNLOAD_LOCATION "${CPM_SOURCE_CACHE}/cpm/CPM_${CPM_DOWNLOAD_VERSION}.cmake")
+elseif(DEFINED ENV{CPM_SOURCE_CACHE})
+  set(CPM_DOWNLOAD_LOCATION "$ENV{CPM_SOURCE_CACHE}/cpm/CPM_${CPM_DOWNLOAD_VERSION}.cmake")
+else()
+  set(CPM_DOWNLOAD_LOCATION "${CMAKE_BINARY_DIR}/cmake/CPM_${CPM_DOWNLOAD_VERSION}.cmake")
+endif()
+
+if(NOT (EXISTS ${CPM_DOWNLOAD_LOCATION}))
+  message(STATUS "Downloading CPM.cmake to ${CPM_DOWNLOAD_LOCATION}")
+  file(DOWNLOAD
+       https://github.com/TheLartians/CPM.cmake/releases/download/v${CPM_DOWNLOAD_VERSION}/CPM.cmake
```

Comment:
I know this is the prescribed way of doing this in CPM, but using a git tag instead of commit hash means that someone could in theory change it out from underneath us. We should use a commit hash instead:

Here's the link to the commit hash at 0.27.3: https://raw.githubusercontent.com/TheLartians/CPM.cmake/c3c7e2d9a3b890511891ffc415d7ea23c689068a/cmake/CPM.cmake

Reply:
I updated the script to pull by hash. The script it pulls actually has `set(CURRENT_CPM_VERSION 0.27.2-development-version)`, but it's otherwise identical to 0.27.3's. I think that's because they change the version info only within their actual release process.

Reply:
Why would they change a release tag in a way that would break users?